BackEnd/src/main/java/com/exadel/carpoolfree/controller/DriveController.java

```diff
@@ -0,0 +1,49 @@
+package com.exadel.carpoolfree.controller;
+
+
+import com.exadel.carpoolfree.model.Drive;
+import com.exadel.carpoolfree.repository.DriveRepository;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/drive")
+public class DriveController {
+    private final DriveRepository driveRepository;
+
+    public DriveController(DriveRepository driveRepository) {
+        this.driveRepository = driveRepository;
+    }
+
+    //Is it need to return absolutely all drives?
+
+    @GetMapping("/{id}")
+    public Drive findById(final @PathVariable Long id) {
+        return driveRepository.findById(id).get();
+    }
+
+
+    @GetMapping("/startTime/{startTime}")
+    public List<Drive> findAllByStartTime(final @PathVariable String startTime) {
+        return driveRepository.findAllByStartTime(startTime);
```

Comment:
Something to consider for the future. It may be ok for now but we need to think how presize we want to be - is it minutes or seconds? I think users would need a time interval to search within.

Reply:
in progress